ee/clickhouse/queries/funnels/base.py

```diff
@@ -0,0 +1,127 @@
+from abc import ABC, abstractmethod
+from typing import List, Tuple
+
+from django.utils import timezone
+
+from ee.clickhouse.client import sync_execute
+from ee.clickhouse.models.action import format_action_filter
+from ee.clickhouse.models.property import parse_prop_clauses
+from ee.clickhouse.queries.util import parse_timestamps
+from ee.clickhouse.sql.person import GET_LATEST_PERSON_DISTINCT_ID_SQL
+from posthog.constants import TREND_FILTER_TYPE_ACTIONS
+from posthog.models import Action, Entity, Filter, Team
+from posthog.models.filters.mixins.funnel_window_days import FunnelWindowDaysMixin
+from posthog.queries.funnel import Funnel
+from posthog.utils import relative_date_parse
+
+
+class ClickhouseFunnelBase(ABC, Funnel):
+    _filter: Filter
+    _team: Team
+
+    def __init__(self, filter: Filter, team: Team) -> None:
+        self._filter = filter
+        self._team = team
+        self.params = {
+            "team_id": self._team.pk,
+            "events": [],  # purely a speed optimization, don't need this for filtering
+        }
+
+    def run(self, *args, **kwargs):
+        if len(self._filter.entities) == 0:
+            return []
+
+        # if self._filter.display == TRENDS_LINEAR:
+        #     return ClickhouseFunnelTrends(self._filter, self._team).run()
+        # else:
```

Comment:
Why is there stuff commented out?

Reply:
It's getting reworked. It won't be there once it's finished.